src/transport/GroupPeerMessageCounter.cpp

```diff
@@ -222,6 +231,28 @@ void GroupPeerTable::CompactPeers(GroupSender * list, uint32_t size)
     }
 }
 
+void GroupPeerTable::RemoveAndCompactFabric(uint32_t tableIndex)
+{
+    if (tableIndex >= CHIP_CONFIG_MAX_FABRICS)
+    {
+        return;
+    }
+    mGroupFabrics[tableIndex].mFabricIndex = kUndefinedFabricIndex;
+    // To maintain logic integrity Fabric array cannot have empty slot in between data
+    // Find the last non empty element
+    for (uint32_t i = CHIP_CONFIG_MAX_FABRICS - 1; i > tableIndex; i--)
```

Comment:
This is a pre-existing issue, so followup ok for it, but I just noticed: what happens if the last element that has `mFabricIndex != kUndefinedFabricIndex` is in fact the element we are trying to remove?  As far as I can tell it won't be removed....

Reply:
good catch